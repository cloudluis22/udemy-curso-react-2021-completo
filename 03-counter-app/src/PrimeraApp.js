import React from 'react';
import PropTypes from 'prop-types';

// Functional Components
// El parametro es una propiedad.
const PrimeraApp = ( { mensaje, subtitulo } ) => {

    // Solo se puede retornar un elemento a la vez, pero usando un fragment puedes adjuntar varios elementos en uno solo.
    return (
        // Esto es un fragment
        <> 
            <h1>{mensaje}</h1>
            <p>{subtitulo}</p>
        </>
    );


}

// Los PropTypes sirven para hacer requeridas las propiedades y solo de un cierto tipo al usar un componente.
PrimeraApp.propTypes = {
    mensaje: PropTypes.string.isRequired // la propiedad "mensaje" requiere ser un string y es obligatorio usarlo.
}

// Los default props sirven para dar valores por defecto a las propiedades si no se les asigna uno al ser utilizados.
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;