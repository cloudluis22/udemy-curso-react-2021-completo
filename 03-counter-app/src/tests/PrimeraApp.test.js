import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en el componente de PrimeraApp.', () => {

    test('Debe mostrar el componente de PrimeraApp correctamente.', () => {
        
        const wrapper = shallow(<PrimeraApp mensaje="Hola Mundo" />)

        expect(wrapper).toMatchSnapshot();

    });
    
    test('Debe mostrar el subtitulo enviado por props.', () => {
        
        
        const saludo = 'Hola Mundo';
        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow(
            <PrimeraApp
            mensaje = { saludo }
            subtitulo = { subtitulo }
            />
        );

        const parrafo = wrapper.find('p').text();
        
        expect(parrafo).toBe(subtitulo);

    });
    
});