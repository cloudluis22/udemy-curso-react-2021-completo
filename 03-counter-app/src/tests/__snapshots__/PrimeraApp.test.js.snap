// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de PrimeraApp. Debe mostrar el componente de PrimeraApp correctamente. 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
